sovtokenfees/sovtokenfees/test/auth_map/test_auth_rules.py

```diff
@@ -0,0 +1,519 @@
+import json
+import sys
+from typing import NamedTuple, Dict
+
+import pytest
+from indy.did import replace_keys_start
+from indy.ledger import build_attrib_request
+from indy_common.authorize.auth_constraints import AuthConstraint, ROLE, AuthConstraintOr, AbstractAuthConstraint, \
+    IDENTITY_OWNER, AuthConstraintAnd
+
+from indy_node.test.auth_rule.helper import sdk_send_and_check_auth_rule_request
+
+from indy_common.constants import NYM, TRUST_ANCHOR, NODE, POOL_UPGRADE, POOL_RESTART, VALIDATOR_INFO, GET_SCHEMA, \
+    ATTRIB, TRUST_ANCHOR_STRING
+
+from indy_common.authorize.auth_actions import EDIT_PREFIX, ADD_PREFIX
+from libnacl.secret import SecretBox
+from sovtoken.constants import AMOUNT, ADDRESS
+from sovtokenfees.constants import FEES, FEES_FIELD_NAME
+from sovtokenfees.test.helper import add_fees_request_with_address
+
+from plenum.common.constants import TRUSTEE, STEWARD, STEWARD_STRING, TRUSTEE_STRING, VERKEY
+from plenum.common.exceptions import RequestRejectedException
+from plenum.test.helper import sdk_multisign_request_object, sdk_multi_sign_request_objects, sdk_json_to_request_object
+from plenum.test.pool_transactions.helper import sdk_add_new_nym
+
+auth_constraint = AuthConstraint(role=TRUSTEE, sig_count=1, need_to_be_owner=False)
+
+fee_1 = ("1", 1)
+fee_2 = ("2", 2)
+fee_3 = ("3", 3)
+fee_5 = ("5", 5)
+fee_6 = ("6", 6)
+fee_100 = ("100", 100)
+
+set_fees = dict([
+    fee_1,
+    fee_2,
+    fee_3,
+    fee_5,
+    fee_6,
+    fee_100
+])
+
+RequestParams = NamedTuple("RequestParams", [("fees", int),
+                                             ("owner", str),
+                                             ("wallets", Dict[str, int])]
+                           )
+RequestParams.__new__.__defaults__ = (0, "-1", {})
+
+InputParam = NamedTuple("InputParam", [
+    ("auth_constraint", AbstractAuthConstraint),
+    ("valid_requests", [RequestParams]),
+    ("invalid_requests", [RequestParams])])
+
+steward_address = ""
+trustee_address = ""
+owner_address = ""
+
+input_params_map = [
+    # 0
+    InputParam(auth_constraint=AuthConstraintOr([AuthConstraint(STEWARD, 1,
+                                                                metadata={FEES_FIELD_NAME: fee_5[0]}),
+                                                 AuthConstraint(TRUSTEE, 1)]),
+               valid_requests=[
+                   RequestParams(fees=fee_5[1],
+                                 wallets={STEWARD: 1}),
+                   RequestParams(fees=0,
+                                 wallets={TRUSTEE: 1}),
+                   RequestParams(fees=fee_5[1],
+                                 wallets={TRUSTEE: 1,
+                                          STEWARD: 1}),
+               ],
+               invalid_requests=[
```

Comment:
Please consider cases where we send incorrect number of Fees (in this an other test cases):
- 0
- None (no fees at all)
- fee_5[1] - 1
- fee_5[1] + 1